erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -1566,6 +1566,37 @@ func (hph *HexPatriciaHashed) createCellGetter(b []byte, updateKey []byte, row,
 	}
 }
 
+// updateKind is a type of update that is being applied to the trie structure.
+type updateKind uint8
+
+const (
+	// updateKindDelete means after we processed longest common prefix, row ended up empty.
+	updateKindDelete updateKind = 0b0
+
+	// updateKindPropagate is an update operation ended up with a single nibble which is leaf or extension node.
+	// We do not store keys with only one cell as a value in db, instead we copy them upwards to the parent branch.
+	// IF updated prefix was present before, operation produces a delete write for it.
+	updateKindPropagate updateKind = 0b01
+
+	// updateKindBranch is an update operation ended up as a branch of 2+ cells.
+	// That does not necessarily means that branch is NEW, it could be an existing branch that was updated.
+	updateKindBranch updateKind = 0b10
+)
+
+// Kind defines how exactly given update should be folded upwards to the parent branch or root.
+// It also returns number of nibbles that left in branch after the operation.
```

Comment:
If second return value is indeed the number of nibbles that are present in the branch after the update then why is the variable called `nibblesUpdated`  (which would suggest this is the number of nibbles that were updated)?